tests/ts_cluster_updowngrade_select_aggregation_SUITE.erl

```diff
@@ -0,0 +1,205 @@
+%% -------------------------------------------------------------------
+%%
+%% Copyright (c) 2016 Basho Technologies, Inc.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+%% -------------------------------------------------------------------
+-module(ts_cluster_updowngrade_select_aggregation_SUITE).
+
+-ifdef(EQC).
+-include_lib("eqc/include/eqc.hrl").
+
+-export([
+         suite/0,
+         init_per_suite/1,
+         end_per_suite/1,
+         all/0,
+         run_this_test/1
+        ]).
+
+%% EQC exports
+-export([gen_scenario/0, gen_version/0, prop_scenario/1]).
+
+-include_lib("common_test/include/ct.hrl").
+-include("ts_updown_util.hrl").
+
+-define(TABLE, "updown_aggregation_test_table").
+-define(TEMPERATURE_COL_INDEX, 4).
+-define(PRESSURE_COL_INDEX, 5).
+-define(PRECIPITATION_COL_INDEX, 6).
+
+%% Callbacks
+
+suite() ->
+    [{timetrap, {seconds, 9000}}].
+
+all() ->
+    [
+     run_this_test
+    ].
+
+init_per_suite(Config) ->
+    lists:foldl(
+      fun(Fun, Cfg) -> Fun(Cfg) end,
+      Config,
+      [fun ts_updown_util:setup/1,
+       fun make_scenario_invariants/1]).
+
+end_per_suite(Config) ->
+    ts_updown_util:maybe_shutdown_client_node(Config).
+
+
+%% it's 2 to the power of the number of binary scenario parameters in
+%% #scenario{} (6)
+-define(NUMCASES, 64).
+
+run_this_test(Config) ->
+    case eqc:quickcheck(
+           eqc:numtests(?NUMCASES, prop_scenario(Config))) of
+        true ->
+            pass;
+        _ ->
+            ct:fail("There were failing queries", [])
+    end.
+
+prop_scenario(Config) ->
+    ?FORALL(
+       Sce, gen_scenario(),
+       [] == ts_updown_util:run_scenario(Config, Sce)).
+
+gen_scenario() ->
```

Comment:
The generator I wrote in Ukraine had left and stopped states - so it would test a range of other ring states - these have been dropped in this implementation...